### PR TITLE
fix: pass url to contents.hasServiceWorker()

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

A bug was present in the original API, wherein the API assumed that the loaded page has a previously committed url and thus would DCHECK when a new page was loaded. Thus, for this to actually work properly we need to pass in the url to check as a new parameter. This PR thus does that and updates the associated tests and documentation.

cc @deepak1556 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed `webContents.hasServiceWorker` by adding a parameter for the url to test for its presence.

### CODE REVIEW

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -1448,13 +1448,8 @@ v8::Local<v8::Promise> WebContents::HasServiceWorker() {
...
 
   GURL url = web_contents()->GetLastCommittedURL();
-  if (!url.is_valid()) {
-    promise->RejectWithErrorMessage("URL invalid or not yet loaded.");
-    return promise->GetHandle();
-  }
-
   context->CheckHasServiceWorker(
-      url, url, base::BindOnce(&OnServiceWorkerCheckDone, promise));
+      url, other_url, base::BindOnce(&OnServiceWorkerCheckDone, promise));
```
[the remark this answers is not in the document]
Comment (deepak1556) [reply atom/browser/api/atom_api_web_contents.cc]: Agree with @ckerr , thats the correct way to query this api.

[the remark this answers is not in the document]
Comment (codebytere) [reply atom/browser/api/atom_api_web_contents.cc]: updating 🙇‍♀️

File: spec/package-lock.json [hunk 2]
```diff
@@ -228,7 +228,7 @@
     },
     "commander": {
       "version": "2.15.1",
-      "resolved": "http://registry.npmjs.org/commander/-/commander-2.15.1.tgz",
```
Comment (ckerr) [inline spec/package-lock.json]: Was `package.json` included by accident?

**Comment by ckerr** [comment]
readding `semver/major` for now because this introduces a breaking change from the 4-0-x API as documented [here](https://github.com/electron/electron/blob/4-0-x/docs/api/web-contents.md#contentshasserviceworkercallback).

I agree that the bug that caused a `DCHECK()` makes this less-than-useful in 4-0-x and from that we could argue that this isn't a breaking change, but since we ship with `DCHECK()` disabled I'd like to know what the actual end-user behavior was before. If users were actually able to use this API in 4.0 then this is a breaking change IMO.

**Comment by codebytere** [comment]
Likely closing in light of https://github.com/electron/electron/pull/16717.

**Pull request closed without merging**